.github/workflows/docs.yml

```
@@ -15,6 +15,9 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
+env:
+  VTK_DEFAULT_OPENGL_WINDOW: "vtkEGLRenderWindow"
```

Is there a way to define `"vtkEGLRenderWindow"` as a variable in tox for CI use only, such that it can be reused, e.g. for the docstring/doctest workflow? I.e. address this comment https://github.com/pyvista/pyvista/pull/7861#issuecomment-3212292384

Maybe define `CI_VTK_DEFAULT_OPENGL_WINDOW = "vtkEGLRenderWindow"` as the variable in tox, and do something like:
```suggestion
  VTK_DEFAULT_OPENGL_WINDOW: CI_VTK_DEFAULT_OPENGL_WINDOW
```

---

> Is there a way to define `"vtkEGLRenderWindow"` as a variable in tox for CI use only, such that it can be reused, e.g. for the docstring/doctest workflow? I.e. address this comment [#7861 (comment)](https://github.com/pyvista/pyvista/pull/7861#issuecomment-3212292384)
> 

Don't think it's possible to set and `env` variable in `tox` depending on the value of another one (ie. CI)

> alternatively, define a pre-command that checks if the env var `CI` is true, and if it is then export `VTK_DEFAULT_OPENGL_WINDOW = "vtkEGLRenderWindow"`? This way we wouldn't need to define it as an env var inside the workflow itself, and would be self-contained inside the tox config.

commands output cannot be captured in subsequent commands, so I don't think this is possible either.

Maybe through a dedicated homemade plugin ...

---

Does https://github.com/pyvista/pyvista/pull/7945/commits/f7c8862ba784ad93fb0f2870b0b201cf5f9c6ad4 is what you intended ?

If yes, do you prefer cherry-picking https://github.com/pyvista/pyvista/pull/7945/commits/f7c8862ba784ad93fb0f2870b0b201cf5f9c6ad4 inside another dedicated PR ?

---

Maybe we set a new env var `PYVISTA_CI=true` in tox, and set the opengl window var in pyvista's init before any vtk imports?

``` py
if os.environ.get("PYVISTA_CI", "").lower() == "true":
    os.environ["VTK_DEFAULT_OPENGL_WINDOW"] = "vtkEGLRenderWindow"
```
But for now it's OK, we can try this (or not) in the docstring workflow PR when there's an actual duplication of variables

---

> @user27182 does [f7c8862](https://github.com/pyvista/pyvista/commit/f7c8862ba784ad93fb0f2870b0b201cf5f9c6ad4) is what you intended ?
> 
> If yes, do you prefer cherry-picking [f7c8862](https://github.com/pyvista/pyvista/commit/f7c8862ba784ad93fb0f2870b0b201cf5f9c6ad4) inside another dedicated PR ?

I think this works? Not at all familiar with tox. If it lets us define it exactly once in tox and reuse it in other workflows as needed then that's goal achieved.

I have no preference for adding this as part of this PR or as part of a future docstring workflow tox PR.

---

> docstring are already tested with tox

I see, then I guess we should include `doctest-modules` in the list of cases where we want to set `VTK_DEFAULT_OPENGL_WINDOW`? Is that correct, @akaszynski, we want these two workflows to use this consistently?

And I went to verify it's working by checking the software report, but I guess there is none?? Or at least I couldn't find it. Maybe it was excluded with #7916 ?

And as an additional unrelated note, is it an easy change to _exclude_ `-v` from the documentation image tests? I find the extra 5000 lines in the log for all the test cases to be unnecessary for this. (`-v` was not used for this prior to #7916)

---

> I was expecting the report to include "vtkEGLRenderWindow" as the render window

This is now working, and "vtkEGLRenderWindow" is reported by the build
https://github.com/pyvista/pyvista/actions/runs/17700376046/job/50305304861?pr=7945#step:7:41
and the docstring tests
https://github.com/pyvista/pyvista/actions/runs/17700376033/job/50305304850?pr=7945#step:7:41

Just need to determine whether we want this render window set just for these two jobs or if we want this for all CI jobs, including unit tests.